framework/src/state_machine/state_machine.ts

```diff
@@ -165,7 +165,7 @@ export class StateMachine {
 		}
 		const command = this._getCommand(ctx.transaction.moduleID, ctx.transaction.commandID);
 		// Execute command
-		ctx.eventQueue.createSnapshot();
+		const snapshotID = ctx.eventQueue.createSnapshot();
 		ctx.stateStore.createSnapshot();
```

Comment:
inside of `createSnapshot` it should be calling stateReadWriter.createSnapshot and that should return `id` ad well similar to event queue

Reply:
Should stateReadWriter be added to the ctx?

Reply:
the `stateStore` is `PrefixedStateReadWriter`, so [here](https://github.com/LiskHQ/lisk-sdk/blob/b93207dbd1cd34b2fbe016f57c43c139873754fc/framework/src/state_machine/prefixed_state_read_writer.ts#L120-L127) i think you need to get the snapshotID and use them

Reply:
Updated!